vulnerabilitylookupsighting/publish.py

```diff
@@ -21,61 +19,37 @@ def remove_case_insensitive_duplicates(input_list: list[str]) -> list[str]:
 def push_sighting_to_vulnerability_lookup(attribute, vulnerability_ids):
     """Create a sighting from a MISP attribute and push it to the Vulnerability Lookup instance."""
     print("Pushing sightings to Vulnerability Lookup...")
-    headers_json = {
-        "Content-Type": "application/json",
-        "accept": "application/json",
-        "X-API-KEY": f"{config.vulnerability_auth_token}",
-    }
+    vuln_lookup = PyVulnerabilityLookup(config.vulnerability_lookup_base_url, token=config.vulnerability_auth_token)
     for vuln in vulnerability_ids:
         # Set the creation timestamp for the sighting
         creation_timestamp = ""
-        if attribute.get("first_seen", False):
+        if hasattr(attribute, "first_seen") and attribute.first_seen:
             try:
-                creation_timestamp = datetime.fromtimestamp(
-                    int(attribute["first_seen"]), timezone.utc
-                ).isoformat()
+                creation_timestamp = attribute.first_seen.isoformat()
             except Exception:
-                print(
-                    "Bad timestamp: {} for event {}".format(
-                        attribute["first_seen"], attribute["Event"]["uuid"]
-                    )
-                )
+                print(f"Bad timestamp: {attribute.first_seen} for event {attribute.uuid}")
                 continue
         else:
             try:
-                creation_timestamp = datetime.fromtimestamp(
-                    int(attribute["timestamp"]), timezone.utc
-                ).isoformat()
+                creation_timestamp = attribute.timestamp.isoformat()
             except Exception:
-                print(
-                    "Bad timestamp: {} for event {}".format(
-                        attribute["timestamp"], attribute["Event"]["uuid"]
-                    )
-                )
+                print(f"Bad timestamp: {attribute.timestamp} for event {attribute.uuid}")
                 continue
         if not creation_timestamp:
             continue
 
         # Create the sighting
         sighting = {
             "type": "seen",
-            "source": f"MISP/{attribute['Event']['uuid']}",
+            "source": f"MISP/{attribute.uuid}",
```

Comment:
@Rafiot  actually here the goal was to have the UUID of the event. I guess attribute.uuid is the UUID of the attribute no ?
for example [here](https://vulnerability.circl.lu/sightings/?query=MISP) it's UUID of events.

Reply:
oh, right, my bad, this should be `attribute.Event.uuid`, let me check if I'm correct...